lib/chopperbot/split/parser.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Chopperbot.Split.Parser do
+  alias Chopperbot.Split.{
+    InputTransformer,
+    OptionTransformer,
+    Order
+  }
+
+  @white_space_pattern ~r/\p{Zs}/u
+
+  @type parsed :: %{orders: [Order.t()], multiplier: float()}
+
+  @doc """
+  Parse text into the correct orders & multiplier.
+
+  ## Examples
+      iex> parse("a 100 a 200 b 300 +v +s")
+      {:ok, %{orders: [{"a", 100.0}, {"a", 200.0}, {"b", 300.0}], multiplier: 1.177}}
+
+      iex> parse("a 100 b 300 -50%")
+      {:ok, %{orders: [{"a", 100.0}, {"b", 300.0}], multiplier: 0.5}}
+
+      iex> parse("a 100 b 300 +t +invalid")
+      {:error, :invalid_option, ["+t", "+invalid"]}
+
+      iex> parse("a one-hundred b ten baht +v +s")
+      {:error, :invalid_input, ["one-hundred", "ten", "baht"]}
+  """
+  @spec parse(String.t()) ::
+          {:ok, parsed()}
```

Comment:
I am thinking about `parsed` not sure if abstracting the type into `parsed` worth sacrificing the type reading, since `parsed` is not reused that much too.

Reply:
> I felt like `transform_to_orders` is still a little hard to read. Let me come back to think again tomorrow!

gracies 🙇

Reply:
it's used only once tho lol, can be replaced with just `%{orders: [Order.t()], multiplier: float()}` 🙇